_format/1.1/index.md

```diff
@@ -1834,8 +1909,575 @@ An error object **MAY** have the following members:
 [resource links]: #document-resource-object-links
 [resource identifier object]: #document-resource-identifier-objects
 [compound document]: #document-compound-documents
+[identities]: #document-identities-array
 [meta]: #document-meta
 [links]: #document-links
 [error details]: #errors
 [member names]: #document-member-names
 [pagination]: #fetching-pagination
+
+## <a href="#operations" id="operations" class="headerlink"></a> Operations
+
+A server **MAY** support usage of the `operations` array to complete one or more
+fetch or mutation operations in a single request.
+
+All HTTP requests that include an `operations` array **MUST** be issued with
+`PATCH`.
+
+For the sake of clarity, requests that include operations are called
+"operations requests" while requests that perform a single fetch or mutation are
+called "singular requests".
+
+### <a href="#operation-objects" id="operation-objects" class="headerlink"></a> Operation Objects
+
+An operation object **MAY** contain any of the following members:
+
+* `op`: an operation code, expressed as a string, that indicates the type of
+  operation to perform. Valid values include:
+  - `"get"`: fetches a resource, resource identity, collection of resources, or
+     collection of resource identitities
+  - `"add"`: creates a new resource or relationship
+  - `"replace"`: updates a resource or relationship
+  - `"remove"`: removes a resource or relationship
+
+* `data`: the operation's "primary data".
+
+* `included`: an array of [resource
+  objects](#document-resource-objects) that are
+  related to the operation's primary data and/or each other
+  ("included resources").
+
+* `ref`: the resource or resource collection that is the target of the
+  operation. This member serves to disambiguate the target at a particular
+  endpoint if that endpoint represents multiple resources. `ref` **MUST** be
+  expressed as an object that **MAY** contain any of the following members:
+  - `type`: resource `type`
+  - `id` or `lid`: resource identifier or local identifier
+  - `relationship`: the name of the relationship
+
+* `params`: all of the query parameters allowed for singular requests can be
+  represented in JSON form in an operation's `params` object. Possible members
+  include:
+  - `include`
+  - `sort`
+  - `filter`
+  - `page`
+  - `fields`
+
+* `links`: a [links object](#document-links) related to the operation's primary
+  data.
+
+* `meta`: a [meta object](#document-meta) that contains non-standard
+  meta-information about the operation.
+
+Different members are required for processing different types of operations, as
+described below.
+
+> Note: `op` codes are used instead of the HTTP methods prescribed for singular
+  requests. The values `"get"`, `"add"`, `"replace"`, and `"remove"` align with
+  the HTTP methods `GET`, `POST`, `PATCH`, and `DELETE` used for singular
+  requests. The specific `op` codes are inspired by [RFC
+  6902](https://tools.ietf.org/html/rfc6902).
```

Comment:
I don't love the choice of `replace` here. In the referenced JSON-patch spec, `replace` is only ever applied to a single field, replacing it entirely. In this spec, it's applied to a document, but has the semantics of an update - keys not mentioned in the `replace` op are not modified. `replace` seems to imply the whole record would be replaced with only the new values.
Could we change to `patch` or `update`? I think otherwise it will cause confusion for people learning the spec.

Reply:
I'll second that sentiment. Until @courajs just pointed out the update semantics, I fell into the trap thinking "replace" _was_ like a PUT.

Reply:
I agree that `replace` could cause confusion and will probably change it to `update`. I'm trying to avoid usage of HTTP methods, since operations exist at a different layer, so I'm reluctant to use `patch`.

Without perfect alignment, I should probably do away with the reference to RFC 6902 as well.